src/executor/evaluate/mod.rs

```diff
@@ -408,11 +408,42 @@ async fn evaluate_function<'a, T: 'static + Debug>(
         Function::Pi() => {
             { Ok(Evaluated::from(Value::F64(std::f64::consts::PI))) }.map(Evaluated::from)
         }
-        Function::Trim(expr) => match eval_to_str(expr).await? {
-            Nullable::Value(string) => Ok(Value::Str(string.trim().to_owned())),
-            Nullable::Null => Ok(Value::Null),
+        Function::Trim { expr, trim_where } => {
+            let expr_str = match eval_to_str(expr).await? {
+                Nullable::Value(str) => str,
+                Nullable::Null => return Ok(Value::Null).map(Evaluated::from),
+            };
+            let expr_str = expr_str.as_str();
+
+            if expr_str.is_empty() {
+                return Ok(Value::Str(expr_str.to_owned())).map(Evaluated::from);
+            }
+
+            let (trim_where_field, filter_chars) = match trim_where {
+                Some(trim_where) => trim_where,
+                None => {
+                    return Ok(Evaluated::from(Value::Str(expr_str.trim().to_owned())));
+                }
+            };
+
+            let filter_chars = match filter_chars {
+                Some(expr) => match eval_to_str(expr).await? {
+                    Nullable::Value(str) => str.chars().collect::<Vec<_>>(),
+                    Nullable::Null => return Ok(Evaluated::from(Value::Null)),
+                },
+                None => vec![' '],
```

Comment:
These two cases need to be covered in the integration tests.

Reply:
To test this you need to modify sqlparser.

Reply:
Oh, I see.
Is there a reason to change filter_chars type from `Expr` to `Option<Expr>`?
It looks like that conversion is not necessary.
If we keep `Expr` then,
```rust
let filter_chars = match eval_to_str(filter_chars).await? {   
    Nullable::Value(str) => str.chars().collect::<Vec<_>>(),  
    Nullable::Null => return Ok(Evaluated::from(Value::Null)),
};                                                            
```
No unreachable case is required to be covered.

Reply:
This is to add only translate and error cases after modification in sqlparser in the future.

If you feel you don't need it, I'll restore it.

Reply:
Ok,
At first, we don't need to prepare future changes which cannot be covered by us.
Second, I think this is why I was confused.
If we aim to set the proper AST, then... 

Not this,
```rust
Trim {                                         
    expr: Expr,                                
    trim_where: Option<(TrimWhereField, Expr)>,
},       
```           

Here would be the right one..?
```rust
Trim {                                         
    expr: Expr,                                
    filter_chars: Option<Expr>,
    trim_where: Option<TrimWhereField>,
},       
```
Because `filter_chars` and `trim_where` are independent to each other.

Reply:
I think that's right. I relied too much on existing types

Reply:
It's fully understandable. We really know that there were lots of changes you've done on `TRIM`, not only the changes in here, but also contributions to the `sqlparser-rs`. :)

Reply:
Fixed it